### PR TITLE
Add EIP-5450: EOF - Stack Validation

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: EIPS/eip-xxxx.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+eip: xxxx
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:2]:
```suggestion
eip: 5450
```

File: EIPS/eip-xxxx.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+title: EOF - Stack Validation
+description: Deploy-time validation of stack usage for EOF functions.
+author: Andrei Maiboroda (@gumb0)
+discussions-to:
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:6]: Must be thread on ethereum magicians

File: EIPS/eip-xxxx.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+## Rationale
+
+TBA
+
+### Unreachable code
+
+### Clean stack upon termination
+
+## Reference Implementation
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:57]: Should be after backwards compatibility

File: EIPS/eip-xxxx.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    return max_stack_height
+```
+
+## Backwards Compatibility
+
+This change poses no risk to backwards compatibility, as it is introduced only for EOF1 contracts, for which deploying undefined instructions is not allowed, therefore there are no existing contracts using these instructions. The new instructions are not introduced for legacy bytecode (code which is not EOF formatted).
+
+## Security Considerations
+
+TBA
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:145]:
```suggestion
Needs discussion.
```
Personal preference for this section over TBA or TBD.

File: EIPS/eip-xxxx.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+## Copyright
+
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:149]:
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

File: EIPS/eip-xxxx.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+Given new deploy-time guarantees, EVM implementation is not required anymore to have run-time stack underflow check for each executed instruction.
+
+Stack overflow check, on the other hand, is still required at run-time, because function execution can start at arbitrary (i.e. known only at run-time) stack height at CALLF instruction of a caller (i.e. each execution can be in arbitrary inner call frame). Verification alrogirthm examines only stack height changes relative to starting stack height of the function.
+
+## Rationale
+
+TBA
+
+### Unreachable code
+
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:54]:
```suggestion

TBA

```

File: EIPS/eip-xxxx.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+### Clean stack upon termination
```
Comment (Pandapip1) [inline EIPS/eip-xxxx.md:55]:
```suggestion
### Clean stack upon termination

TBA

```

**Review by Pandapip1 — COMMENTED** [review]
There are a few too many placeholders right now. Please either fill in a bit of the motivation or replace it entirely with a `TBA`

File: EIPS/eip-xxxx.md (hunk 6 above)
Comment (gumb0) [reply EIPS/eip-xxxx.md]: Replaced entire Rationale contents with TBA for now.

File: EIPS/eip-5450.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+Stack overflow check, on the other hand, is still required at run-time, because function execution can start at arbitrary (i.e. known only at run-time) stack height at CALLF instruction of a caller (i.e. each execution can be in arbitrary inner call frame). Verification alrogirthm examines only stack height changes relative to starting stack height of the function.
+
+## Rationale
+
+TBA
+
+## Backwards Compatibility
+
+This change poses no risk to backwards compatibility, as it is introduced only for EOF1 contracts, for which deploying undefined instructions is not allowed, therefore there are no existing contracts using these instructions. The new instructions are not introduced for legacy bytecode (code which is not EOF formatted).
```
Comment (Pandapip1) [inline EIPS/eip-5450.md:55]: Note: mention that this requires a hard fork

File: EIPS/eip-5450.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+    if max_stack_height > 1024:
+        raise ValidationException("max stack above limit")
+
+    return max_stack_height
+```
+
+## Security Considerations
+
+TBA
```
Comment (Pandapip1) [inline EIPS/eip-5450.md:141]:
```suggestion
Needs discussion
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-5450.md (hunk 9 above)
Comment (gumb0) [reply EIPS/eip-5450.md]: Changed.

File: EIPS/eip-5450.md (hunk 8 above)
Comment (gumb0) [reply EIPS/eip-5450.md]: Added.

**Review by Pandapip1 — APPROVED** [review]
Seems to meet the basic requirements for a draft.

**Pull request merged**